// ngx-simple-countdown/esm2020/lib/ngx-simple-timeago.directive.mjs
import { Directive, Input } from '@angular/core';
import { interval, ReplaySubject } from 'rxjs';
import { tap, takeUntil, filter } from 'rxjs/operators';
import { getCountdownResult, getDateNow, getLanguage } from './utils';
import * as i0 from "@angular/core";
export class NgxSimpleTimeagoDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.language = 'en';
        this.reactive = true;
    }
    ngOnInit() {
        this.dateNow = getDateNow();
        this.keywords = getLanguage(this.language);
        this.initSimpleCountdown();
    }
    ngOnDestroy() { }
    initSimpleCountdown() {
        this.totalSecondes = this.dateNow - this.date;
        this.updateSimpleCountdown(this.totalSecondes);
        this.interval$ = interval(20000).pipe(takeUntil(this.componentDestroyed(this)), filter(() => this.reactive && this.totalSecondes > 0), tap(() => {
            this.totalSecondes += 20;
            this.updateSimpleCountdown(this.totalSecondes);
        }));
        if (this.reactive) {
            this.interval$.subscribe();
        }
    }
    createHTML(data) {
        const { day, hours, minutes } = data;
        let o = '';
        if (this.totalSecondes > 0) {
            if (this.totalSecondes <= 60) {
                o += this.keywords.now;
            }
            else {
                o += `${this.keywords.timeago} `;
                if (data.day > 0) {
                    o += `${data.day}${this.keywords.day} `;
                }
                if (hours > 0 || day > 0) {
                    o += `${hours}${this.keywords.hours} `;
                }
                if (minutes > 0 || hours > 0 || day > 0) {
                    o += `${minutes}${this.keywords.minutes} `;
                }
            }
        }
        this.elementRef.nativeElement.innerHTML = o;
    }
    updateSimpleCountdown(secondes) {
        const countdownResult = getCountdownResult(secondes);
        this.createHTML(countdownResult);
    }
    componentDestroyed(component) {
        const oldNgOnDestroy = component.ngOnDestroy;
        const destroyed$ = new ReplaySubject(1);
        component.ngOnDestroy = () => {
            oldNgOnDestroy.apply(component);
            destroyed$.next(undefined);
            destroyed$.complete();
        };
        return destroyed$;
    }
}
NgxSimpleTimeagoDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleTimeagoDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
NgxSimpleTimeagoDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.3.9", type: NgxSimpleTimeagoDirective, selector: "[simpleTimeago]", inputs: { date: "date", language: "language", reactive: "reactive" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleTimeagoDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[simpleTimeago]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { date: [{
                type: Input
            }], language: [{
                type: Input
            }], reactive: [{
                type: Input
            }] } });
