// ngx-simple-countdown/esm2020/lib/ngx-simple-countdown.module.mjs
import { NgModule } from "@angular/core";
import { NgxSimpleCountdownDirective } from "./ngx-simple-countdown.directive";
import { NgxSimpleTimeagoDirective } from "./ngx-simple-timeago.directive";
import * as i0 from "@angular/core";
export class NgxSimpleCountdownModule {
}
NgxSimpleCountdownModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleCountdownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxSimpleCountdownModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleCountdownModule, declarations: [NgxSimpleCountdownDirective, NgxSimpleTimeagoDirective], exports: [NgxSimpleCountdownDirective, NgxSimpleTimeagoDirective] });
NgxSimpleCountdownModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleCountdownModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.9", ngImport: i0, type: NgxSimpleCountdownModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxSimpleCountdownDirective, NgxSimpleTimeagoDirective],
                    imports: [],
                    exports: [NgxSimpleCountdownDirective, NgxSimpleTimeagoDirective]
                }]
        }] });
